app/services/export/CalendarExporter.java

```diff
@@ -0,0 +1,153 @@
+  private static final String DATE_STAMP = "DTSTAMP:";  // TODO format: 20060812T125900Z
+
+  private static final Map<String, String> mFieldMap = new HashMap<>();
+  static{
+    mFieldMap.put(UID, JsonLdConstants.ID);
+    mFieldMap.put(COMMON_NAME, "organizer.name.@value");
+    mFieldMap.put(SUMMARY, "name.@value");
+    mFieldMap.put(DESCRIPTION, "description.@value");
+    mFieldMap.put(URL, "url");
+    mFieldMap.put(DATE_START, "startDate");
+    mFieldMap.put(DATE_END, "endDate");
+    mFieldMap.put(LOCATION, "location.address.streetAddress".concat(VALUE_SEPARATOR)
+                    .concat("location.address.postalCode").concat(VALUE_SEPARATOR)
+                    .concat("location.address.addressLocality").concat(VALUE_SEPARATOR)
+                    .concat("location.address.addressRegion").concat(VALUE_SEPARATOR)
+                    .concat("location.address.addressCountry").concat(VALUE_SEPARATOR));
+
+    mFieldMap.put(GEO, "location.geo.lat".concat(VALUE_SEPARATOR).concat("location.geo.lon"));
+  }
+
+  private final Locale mPreferredLocale;
+
+  public CalendarExporter(Locale aPreferredLocale){
+    mPreferredLocale = aPreferredLocale;
+  }
+
+  @Override
+  public String export(Resource aResource) {
+    if (!aResource.getType().equals("Event")) {
+      return null;
+    }
+    StringBuilder result = new StringBuilder(HEADER);
+    result.append(exportResourceWithoutHeader(aResource));
+    result.append(FOOTER);
+    return result.toString();
+  }
+
+  @Override
+  public String export(ResourceList aResourceList) {
+    StringBuilder result = new StringBuilder(HEADER);
+    for (Resource resource : aResourceList.getItems()){
+      if (resource.getType().equals("Event")){
+        result.append(exportResourceWithoutHeader(resource));
+      }
+    }
+    result.append(FOOTER);
+    return result.toString();
+  }
+
+  private String exportResourceWithoutHeader(Resource aResource){
+    StringBuilder result = new StringBuilder(EVENT_BEGIN);
+    for (Map.Entry<String, String> mapping : mFieldMap.entrySet()){
+      boolean hasAppendedSomething = false;
+      String[] mappingValues = mapping.getValue().split(VALUE_SEPARATOR);
+      for (int i=0; i<mappingValues.length; i++){
+        final Object o = aResource.getNestedFieldValue(mappingValues[i], mPreferredLocale);
+        if (o != null) {
+          String value = o.toString();
+          if (i == 0){
+            result.append(mapping.getKey());
+          } //
+          else{
+            result.append(VALUE_SEPARATOR);
+          }
+          result.append(value);
+          hasAppendedSomething = true;
+        }
+      }
+      if (hasAppendedSomething){
+        result.append("\n");
+      }
+    }
+    result.append(getExportedOrganizer(aResource));
+    result.append(CAL_CLASS);
+    result.append(DATE_STAMP).append(System.currentTimeMillis()).append("\n"); // TODO change time stamp format
```

Comment:
I have pimped the System time in the sense that [Instant](https://docs.oracle.com/javase/8/docs/api/java/time/Instant.html) is used by now and it is formatted as 12345678T123456Z, see https://github.com/hbz/oerworldmap/commit/3037a18ccacf3591959d3b1ec0a2f0a004e64eb7. We can use it like that for now, because the meaning is definitely not wrong. I agree that providing the timestamp of the last edit would be even nicer.